src/gtk/toga_gtk/window.py

```diff
@@ -93,6 +99,10 @@ def set_content(self, widget):
         for child in widget.interface.children:
             child._impl.container = widget
 
+        if is_content_update:
```

Comment:
I'm not sure I follow the purpose of the 'is_content_update' flag. It's only tracking when you *remove* content; and then it forces the window to be shown... are you sure you don't just need to do a rehint and relayout?

Reply:
Your suggestion sounds reasonable, but I'm not sure on which object I should do the rehint/relayout.